styleguide/source/assets/scss/01-atoms/_button.scss

```diff
@@ -2,8 +2,24 @@
 .ama__button,
 button {
   @include ama-button();
-}
 
-.ama__button--secondary {
-  @include ama-button("", "secondary");
+  &--secondary {
+    @include ama-button("", "secondary");
+  }
+
+  &--block {
+    @include ama-button("block", "");
+  }
+
+  &--small {
+    @include ama-button("small", "");
+  }
+
+  &--block, &--secondary {
+    @include ama-button("block", "secondary");
+  }
+
+  &--small, &--secondary {
+    @include ama-button("block", "secondary");
+  }
```

Comment:
I really love the way that you're extending this with just the `&--class` its extremely human readable and really easy to follow!